src/main/java/com/github/terma/jenkins/githubprcoveragestatus/SonarMasterCoverageRepository.java

```diff
@@ -27,10 +29,14 @@
     private final ObjectMapper objectMapper = new ObjectMapper().disable(FAIL_ON_UNKNOWN_PROPERTIES);
     private PrintStream buildLog;
 
-    public SonarMasterCoverageRepository(String sonarUrl, PrintStream buildLog) {
+    public SonarMasterCoverageRepository(String sonarUrl, String sonarToken, PrintStream buildLog) {
         this.sonarUrl = sonarUrl;
         this.buildLog = buildLog;
         httpClient = new HttpClient();
+        httpClient.getState().setCredentials(
+            new AuthScope(null, 0, null, null),
+            new UsernamePasswordCredentials(sonarToken, "")
```

Comment:
So now we are going to have two case with ```sonarToken``` and without. Please make sure that in case without ```sonarToken``` code will work as it was before.